diff --git a/configure b/configure
index ca4c424..8dff1b1 100755
--- a/configure
+++ b/configure
@@ -305,7 +305,7 @@ else
     $echo_n "Checking C compiler... ${nobr}"
     CC_TMP=`($CC --version) 2>/dev/null | head -1`
     case "$CC_TMP" in
-    gcc* | *apple*gcc*) CC_VER=`expr "x$CC_TMP" : 'x.* \(.*\)$'`; echo "gcc $CC_VER" ;;
+    *gcc* | *apple*gcc*) CC_VER=`expr "x$CC_TMP" : 'x.* \(.*\)$'`; echo "gcc $CC_VER" ;;
     icc*) echo icc ;;
     *)  echo "error"
         cat <<-_ERR
@@ -331,7 +331,7 @@ if not $UNSUPPORTED_COMPILER
 then
     $echo_n "Checking C++ compiler... ${nobr}"
     case "$CXX_TMP" in
-    g++* | *apple*g++*) CXX_VER=`expr "x$CXX_TMP" : 'x.* \(.*\)$'`; echo "g++ $CXX_VER" ;;
+    *g++* | *apple*g++*) CXX_VER=`expr "x$CXX_TMP" : 'x.* \(.*\)$'`; echo "g++ $CXX_VER" ;;
     icpc*) echo icpc ;;
     *)  echo "error"
         cat <<-_ERR
@@ -345,7 +345,7 @@ then
 fi
 
 case "$CXX_TMP" in
-g++* | *apple*g++*) CFLAGSEXTRA=-Wall ;;
+*g++* | *apple*g++*) CFLAGSEXTRA=-Wall ;;
 esac
 
 # Auto-vectorization
