From 5bd20e2bf92e21956436695744f73e9e876c6d5b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Valent=C3=ADn=20Kivachuk?= <vk18496@gmail.com>
Date: Mon, 10 Dec 2018 15:29:43 +0100
Subject: [PATCH] fix byte order

---
 src/hardnested/hardnested_bruteforce.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/hardnested/hardnested_bruteforce.c b/src/hardnested/hardnested_bruteforce.c
index 8ad0cf0..ec7b48f 100644
--- a/src/hardnested/hardnested_bruteforce.c
+++ b/src/hardnested/hardnested_bruteforce.c
@@ -169,10 +169,10 @@ crack_states_thread(void* x) {
                 sprintf(progress_text, "Brute force phase completed. Key found: %012" PRIx64, key);
                 if (thread_arg->trgKey == MC_AUTH_A){
                     t.sectors[thread_arg->trgBlock / 4].foundKeyA = true;
-                    memcpy(t.sectors[thread_arg->trgBlock / 4].KeyA, &key, 6);
+                    num_to_bytes(key, 6, t.sectors[thread_arg->trgBlock / 4].KeyA);
                 } else {
                     t.sectors[thread_arg->trgBlock / 4].foundKeyB = true;
-                    memcpy(t.sectors[thread_arg->trgBlock / 4].KeyB, &key, 6);
+                    num_to_bytes(key, 6, t.sectors[thread_arg->trgBlock / 4].KeyB);
                 }
                 hardnested_print_progress(thread_arg->num_acquired_nonces, progress_text, 0.0, 0, thread_arg->trgBlock, thread_arg->trgKey, true);
                 break;
